internal/command/machine/restart.go

```diff
@@ -105,7 +104,11 @@ func runMachineRestart(ctx context.Context) error {
 	for _, machineID := range args {
 		machine, err := flapsClient.Get(ctx, machineID)
 		if err != nil {
-			return fmt.Errorf("could not get machine %s: %w", machineID, err)
+			if strings.Contains(err.Error(), "machine not found") {
+				return fmt.Errorf("could not get machine %s. perhaps this machine doesn't exist, or isn't part of the app '%s'", machineID, app.Name)
```

Comment:
with #1674 being merged (just a few hours ago), it's now possible to get the status code from these errors. I think something like this would work:

```suggestion
			var flapsErr *flaps.FlapsError
			if errors.As(err, &flapsErr) && flapsErr.ResponseStatusCode == http.StatusNotFound {
				return fmt.Errorf("machine %s not found in app '%s'", machineID, app.Name)
			} else if flapsErr != nil {
				return fmt.Errorf("error getting machine %s in app '%s': %w", machineID, app.Name, flapsErr)
```

please feel empowered to merge with this or without this. we can always come back to update this kind of error handling later or in a separate pr.

Reply:
I'll leave this as an enhancement for later, just to get this out the door. that API is really nice to have though, thanks!